pkg/util/credential.go

```diff
@@ -0,0 +1,160 @@
+package util
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	kubeclient "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	clusterv1alpha1 "github.com/karmada-io/karmada/pkg/apis/cluster/v1alpha1"
+	"github.com/karmada-io/karmada/pkg/util/names"
+)
+
+var (
+	clusterResourceKind = clusterv1alpha1.SchemeGroupVersion.WithKind("Cluster")
+)
+
+type generateClusterInControllerPlaneFunc func(controlPlaneConfig, clusterConfig *rest.Config, opts RegistryClusterOption, secret, impersonatorSecret corev1.Secret) (*clusterv1alpha1.Cluster, error)
+
+// ObtainCredentialsFromMemberCluster obtain credentials for member cluster
+func ObtainCredentialsFromMemberCluster(clusterKubeClient kubeclient.Interface, opts RegistryClusterOption) (*corev1.Secret, *corev1.Secret, error) {
+	var err error
+
+	// ensure namespace where the karmada control plane credential be stored exists in cluster.
+	if _, err = EnsureNamespaceExist(clusterKubeClient, opts.ClusterNamespace, opts.DryRun); err != nil {
+		return nil, nil, err
+	}
+
+	// create a ServiceAccount in cluster.
+	serviceAccountObj := &corev1.ServiceAccount{}
```

Comment:
serviceAccountObj will be created only if opts.EnableReportSecret is open.  How about move it to if logic and combine the following if judgments?

Reply:
@Poor12  review again please